charts/seed-controlplane/charts/kube-apiserver/templates/configmap-egress-selection.yaml

```diff
@@ -32,3 +32,32 @@ data:
       connection:
         proxyProtocol: Direct
 {{- end }}
+{{- if .Values.reversedVPN.enabled }}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kube-apiserver-egress-selector-configuration
+  namespace: {{.Release.Namespace}}
+data:
+  egress-selector-configuration.yaml: |-
+    apiVersion: apiserver.k8s.io/v1alpha1
+    kind: EgressSelectorConfiguration
+    egressSelections:
+    - name: cluster
+      connection:
+        proxyProtocol: HTTPConnect
```

Comment:
I now tried to create a K8s v1.17.17 Shoot cluster and the apiserver fails to start with:
```
$ k -n shoot--foo--aws-local logs kube-apiserver-8457d58d64-nzqmw kube-apiserver

Error: failed to validate egress selector configuration: service.connection.type: Unsupported value: "": supported values: "direct", "http-connect"
```

The same error is also reproducible with v1.16.15.

Are you also able to reproduce it in you local setup?

Reply:
Good catch. I will check it locally.
However, I assume this to be related to the `kube-apiserver` proxy support. This is the same reason why konnectivity tunnel was only working with clusters >= 1.18.
With the "pure" open vpn solution, i.e. with a kube-apiserver side-car and without the envoy-proxy side-car of the vpn-seed-server, it should also work with older kubernetes releases.

@ialidzhikov, @mvladev, @rfranzke, @marwinski What do you think? Should we add the "pure" open vpn solution again for older kubernetes releases or simply only enable the new path only for >= 1.18?
I do not know how many old cluster we need to support and how import it is for them to be able to use the reversed tunnel.

Reply:
can't `http-connect` be used for older versions?

Reply:
https://v1-18.docs.kubernetes.io/docs/setup/release/notes/#kube-apiserver-2 mentions that the proxy seems to be new in 1.18. At least, this is how I read it.

Reply:
Yeah, that's probably why the KonnectivityTunnel feature gate was introduced with the following disclaimer.

> Please note this feature is available ONLY for >= 1.18 kubernetes clusters. 

Ref https://github.com/gardener/gardener/pull/2251

Reply:
Ok, I looked a bit deeper. We might be able to make it work with 1.16 and 1.17. These versions still have the egress selector configuration, but slightly different. I will try to make this work.
Unfortunately, 1.15 does not seem to have this. Any suggestions how to handle 1.15 as I heard today that we still support it.

Reply:
I guess it would also be fine to restrict this feature to clusters >= 1.18, similar to the KonnectvityTunnel. We cannot make it work for 1.15 anyways.

Reply:
I made it work with 1.17 and will check 1.16, but that should be the same.
1.15 would be possible with the "pure" open vpn solution, i.e. adding the side-car to the `kube-apiserver`. However, I would only add this complexity if we really need to.

Reply:
New commit adds support for kubernetes 1.16 & 1.17.

Reply:
As I understood that we can live with the restriction of not supporting kubernetes 1.15, I added a corresponding commit to prevent kubernetes clusters with 1.15 and lower to use the reversed vpn falling back to the previous vpn solution.
I also added a note in the revised release note proposal. @rfranzke please update the pull request accordingly.

Reply:
done, thank you!